cpu/cortexm_common/include/cpu.h

```diff
@@ -104,7 +104,8 @@ static inline void cortexm_sleep(int deep)
     }
 
     /* ensure that all memory accesses have completed and trigger sleeping */
-    unsigned state = irq_disable();
+    /* avoid state to be stored in r0 (causes fault in some platforms) */
+    volatile unsigned state = irq_disable();
```

Comment:
maybe make this optional?

Reply:
I see no problem on that, however I cannot see either why this is harmful, and the difference on size is only 4 bytes.

Reply:
well, it is also an extra memory access (vs. register) on every sleep. Don't know if that matters.

Any other opinions?

Reply:
It just makes the variable stored in the stack, so its less than a function call.